src/components/subscriptions/SubscriptionData.jsx

```diff
@@ -28,65 +23,47 @@ export default function SubscriptionData({ children }) {
   const [overview, setOverview] = useState();
   const [users, setUsers] = useState();
   const [searchQuery, setSearchQuery] = useState();
+  const [filter, setFilter] = useState();
 
-  useEffect(
-    () => {
-      Promise.all([
-        fetchSubscriptionDetails(),
-        fetchSubscriptionUsersOverview(),
-        fetchSubscriptionUsers(),
-      ])
-        .then((responses) => {
-          setDetails(responses[0]);
-          setOverview(responses[1]);
-          setUsers(responses[2]);
-        })
-        // eslint-disable-next-line no-console
-        .catch(error => NewRelicService.logAPIErrorResponse(error));
-    },
-    [],
-  );
+  const {
+    fetch, isLoading, errors, data,
+  } = useSubscriptionData({ search: searchQuery, status: filter, page: 1 });
+
+  // Perform network requests
+  useEffect(() => fetch(), [searchQuery, filter]);
+
+  useEffect(() => {
+    if (!isLoading && !errors) {
+      const { subscriptions, overview: subscriptionUsersOverviewData, subscriptionUsers } = data;
+
+      if (subscriptions?.results && subscriptions?.results[0]) {
+        setDetails(subscriptions.results[0]);
+      }
+      setOverview(subscriptionUsersOverviewData);
+      setUsers(subscriptionUsers);
+    }
+  }, [isLoading, errors, data]);
 
   const value = useMemo(
     () => ({
-      details,
+      details: camelCaseObject(details),
```

Comment:
I'd recommend doing the camelCase as soon as we get back the response. For example, here: https://github.com/edx/frontend-app-admin-portal/pull/285/files#diff-04a96483f6eb2c0ed6ca86461790cba5R30

Are all the API responses and their keys now camelCased (beyond just the subscription details response)?

Reply:
Yes, I have converted API responses to came case in the other places that I found